static-assets/components/cstudio-view-controllers/approve.js

```diff
@@ -63,8 +63,8 @@
 
     });
 
-    function getGenDependency() {
-        return genDependency;
+    function getGenDependency(callback) {
+        calculateDependencies(this.submitItems, callback);
```

Comment:
Don't change this but do bear in mind for future dev that returning a promise anywhere that a callback can be needed at some point can avoid this signature change